yoti_python_sdk/dynamic_sharing_service/policy/wanted_attribute_builder.py

```diff
@@ -24,8 +25,35 @@ def with_derivation(self, derivation):
         self.__attribute["derivation"] = derivation
         return self
 
+    def with_accept_self_asserted(self, value=True):
+        """
+        :param value: True if self-asserted details are allowed
+        """
+        self.__attribute["accept_self_asserted"] = value
+        return self
+
+    def with_optional(self, value=True):
+        """
+        :param value: True if this user attribute is optional in the returned receipt
+        """
+        self.__attribute["optional"] = value
+        return self
+
+    def with_constraint(self, constraint):
+        """
+        :param constraint: Adds a constraint (e.g. a source constraint) to the
+                            wanted attribute
+        """
+        if isinstance(constraint, list):
+            self.__constraints.extend(constraint)
+        else:
+            self.__constraints.append(constraint)
+        return self
+
     def build(self):
         """
         :return: The wanted attribute object
         """
-        return self.__attribute.copy()
+        attribute = self.__attribute.copy()
+        attribute["constraints"] = self.__constraints[:]
```

Comment:
@echarrod Should constraints only be included in the payload if they have been added with the builder?

It shouldn't matter if an empty array is provided, so should be fine like this, but wondered if this is something we should keep consistent across SDKs?

Reply:
Ideally yes we'd keep it so it behaves as it currently does without constraints i.e. not including constraints in the payload if they're not specified 👍